sui-execution/latest/sui-adapter/src/execution_engine.rs

```diff
@@ -522,8 +532,65 @@ mod checked {
                     pt,
                 )
             }
-            TransactionKind::AuthenticatorStateUpdate(_) => {
-                todo!()
+            TransactionKind::EndOfEpochTransaction(txns) => {
+                for tx in txns {
+                    match tx {
+                        EndOfEpochTransactionKind::ChangeEpoch(change_epoch) => {
+                            advance_epoch(
+                                change_epoch,
+                                temporary_store,
+                                tx_ctx,
+                                move_vm,
+                                gas_charger,
+                                protocol_config,
+                                metrics.clone(),
+                            )?;
+                        }
+                        EndOfEpochTransactionKind::AuthenticatorStateCreate(create) => {
+                            // This can fail if the framework does not yet have the create()
+                            // function.
+                            // TODO: this is an ephemeral constraint - once all production
+                            // networks have the new framework, we can add an .expect() at the
+                            // end here (because at that point this should only be called during
+                            // genesis of new networks).
+                            setup_authenticator_state_create(
+                                create,
+                                temporary_store,
+                                tx_ctx,
+                                move_vm,
+                                gas_charger,
+                                protocol_config,
+                                metrics.clone(),
+                            )?;
```

Comment:
I thought we need to be able to tolerate the error here? Would the `?` cause early return?

Reply:
you are right - this would have shown up as soon as I tested an upgrade on a local cluster. fixing